components/script/dom/event.rs

```diff
@@ -147,49 +166,195 @@ impl Event {
     pub fn dispatch(
         &self,
         target: &EventTarget,
-        target_override: Option<&EventTarget>,
+        legacy_target_override: bool,
+        // TODO legacy_did_output_listeners_throw_flag for indexeddb
     ) -> EventStatus {
-        assert!(!self.dispatching());
-        assert!(self.initialized());
-        assert_eq!(self.phase.get(), EventPhase::None);
-        assert!(self.GetCurrentTarget().is_none());
-
         // Step 1.
         self.dispatching.set(true);
 
+        warn!("In dispatch, type is {:?}", self.type_());
```

Comment:
This should be a debug! or trace!, right? Same for other new uses in this file.

Reply:
Most of these don't even have a reason to exist now that it generally works; they're just forgotten leftovers.